### PR TITLE
Quantization support for torch.compile

### PR DESCRIPTION
**Pull request by suryasidd**
### Details:
 - Added support for Quantization in torch.compile
 - Removed Torchscript backend
 - Removed Environment variables for setting options

### Tickets:
 - https://jira.devtools.intel.com/browse/CVS-107243
 - https://jira.devtools.intel.com/browse/CVS-131521

### CODE REVIEW

**Comment by suryasidd** [comment]
build_jenkins

**Review by cavusmustafa — APPROVED** [review]

File: .github/workflows/job_python_unit_tests.yml [hunk 1]
```diff
@@ -246,15 +246,6 @@ jobs:
...
 
-      - name: PyTorch torch.compile TORCHSCRIPT Layer Tests
-        if: ${{ fromJSON(inputs.affected-components).PyTorch_FE.test && runner.os != 'macOS' && runner.arch != 'ARM64' }} # Ticket: 126287
-        run: |
-          python3 -m pytest ${LAYER_TESTS_INSTALL_DIR}/pytorch_tests -m precommit_ts_backend --junitxml=${INSTALL_TEST_DIR}/TEST-pytorch.xml
-        env:
-          TEST_DEVICE: CPU
-          TEST_PRECISION: FP32
-          PYTORCH_TRACING_MODE: TORCHSCRIPT
-
```
Comment (rkazants) [inline .github/workflows/job_python_unit_tests.yml:257]: why is it removed?

Comment (suryasidd) [reply .github/workflows/job_python_unit_tests.yml:257]: This is an experimental backend that we created when we just started working on torch fx backend. Now that fx backend is mature we no longer need this backend and haven't been testing any models on it for a while now. So we are removing it.

**Comment by MaximProshin** [comment]
@vshampor , FYI

File: src/bindings/python/src/openvino/frontend/pytorch/torchdynamo/backend_utils.py [hunk 2]
```diff
@@ -63,14 +56,7 @@ def _get_model_caching(options) -> Optional[Any]:
         caching = options["model_caching"]
         if bool(caching) and str(caching).lower() not in ["false", "0"]:
             return True
-        else:
-            return False
-    else:
-        caching = os.getenv("OPENVINO_TORCH_MODEL_CACHING")
```
Comment (mvafin) [inline src/bindings/python/src/openvino/frontend/pytorch/torchdynamo/backend_utils.py:69]: please update docs if you remove env vars https://github.com/openvinotoolkit/openvino/blob/a1de1908bfd5f597d48422f21796daa3bfd08120/docs/articles_en/openvino-workflow/torch-compile.rst#L117

File: src/frontends/pytorch/src/utils_quantize.hpp [hunk 3]
```diff
@@ -144,6 +144,48 @@ Output<Node> quantize(const NodeContext& context,
...
+ * the new QuantizedPtNode.
+ */
+Output<Node> quantize_fx(const NodeContext& context,
+                         const Output<Node>& input,
+                         const Output<Node>& scale,
+                         const Output<Node>& zero_point,
+                         int64_t out_low_i64,
+                         int64_t out_high_i64,
+                         const Output<Node>& quantized_node);
+
```
Comment (mvafin) [inline src/frontends/pytorch/src/utils_quantize.hpp:188]: Do you really need all 4 functions? As I see you only use 2 versions of `quantize_fx`

File: src/bindings/python/src/openvino/frontend/pytorch/torchdynamo/backend_utils.py (hunk 2 above)
Comment (suryasidd) [reply src/bindings/python/src/openvino/frontend/pytorch/torchdynamo/backend_utils.py:69]: Yes maxim, I have told Karol about this already we are going to raise a PR with the docs change

File: .github/workflows/job_python_unit_tests.yml (hunk 1 above)
Comment (suryasidd) [reply .github/workflows/job_python_unit_tests.yml:257]: Thanks maxim. Removed.

File: src/frontends/pytorch/src/utils_quantize.hpp (hunk 3 above)
Comment (suryasidd) [reply src/frontends/pytorch/src/utils_quantize.hpp:188]: Yeah we can add the other functions if needed later. Removing the unused functions

File: src/frontends/pytorch/src/op_table.cpp [hunk 4]
```diff
@@ -946,6 +955,10 @@ const std::map<std::string, CreatorFunction> get_supported_ops_fx() {
         {"get_attr", op::translate_constant},
         {"torchvision.deform_conv2d.default", op::translate_deform_conv},
         {"torchvision.roi_align.default", op::translate_roi_align},
+        {"quantized_decomposed.quantize_per_tensor.default", op::translate_quantize_per_tensor_fx},
+        {"quantized_decomposed.quantize_per_channel.default", op::translate_quantize_per_channel_fx},
```
[the remark this answers is not in the document]
Comment (suryasidd) [reply src/frontends/pytorch/src/op_table.cpp:959]: Hey Roman, For quantization we are still in the preliminary stage, alpha support. We will be doing more testing and adding tests for these in the next release.

**Comment by suryasidd** [comment]
build_jenkins

**Comment by suryasidd** [comment]
build_jenkins

**Review by cavusmustafa — APPROVED** [review]

File: src/frontends/pytorch/src/op_table.cpp [hunk 5]
```diff
@@ -798,6 +802,7 @@ const std::map<std::string, CreatorFunction> get_supported_ops_fx() {
         {"aten.elu_.default", op::inplace_op<op::translate_elu>},
         {"aten.embedding.default", op::translate_embedding},
         {"aten.empty.memory_format", op::translate_empty},
+        // {"aten.empty_like.default", op::translate_empty_like}, - Need further debugging to enable this op
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:805]:
```suggestion
```
Please remove the commented op.

**Review by mvafin — CHANGES_REQUESTED** [review]
Please add a test for this quantize functionality

**Comment by mvafin** [comment]
build_jenkins

**Review by mvafin — APPROVED** [review]
Please add tests in the separate PR

**Pull request merged**